src/components/FixedGrid/Grid.js

```diff
@@ -24,8 +28,121 @@ function Grid(props) {
 
   const gridClassName = classNames(styles.grid, className);
 
+  const handleOnKeyUp = event => {
+    const keycode = event.code;
+    const { columns, rows } = other;
+    if (event.repeat) return;
+
+    const compareKeys = code => {
+      const right = code === 'ArrowRight';
+      const left = code === 'ArrowLeft';
+      const up = code === 'ArrowUp';
+      const down = code === 'ArrowDown';
+
+      if (!(right || left || up || down)) return true;
+      event.preventDefault();
+
+      const rightLimit = focusPosition.x >= columns - 1;
+      const leftLimit = focusPosition.x <= 0;
+      const topLimit = focusPosition.y <= 0;
+      const bottomLimit = focusPosition.y >= rows - 1;
+      if (right) {
+        if (rightLimit) {
+          focusPosition.x = 0;
+          return;
+        }
+        focusPosition.x = focusPosition.x + 1;
+        return;
+      }
+      if (left) {
+        if (leftLimit) {
+          focusPosition.x = columns - 1;
+          return;
+        }
+        focusPosition.x = focusPosition.x - 1;
+        return;
+      }
+      if (up) {
+        if (topLimit) {
+          focusPosition.y = rows - 1;
+          return;
+        }
+        focusPosition.y = focusPosition.y - 1;
+        return;
+      }
+      if (down) {
+        if (bottomLimit) {
+          focusPosition.y = 0;
+          return;
+        }
+        focusPosition.y = focusPosition.y + 1;
+        return;
+      }
+    };
+
+    const isNotArrowKey = compareKeys(keycode);
+    if (isNotArrowKey) return;
+
+    const currentId = `${focusPosition.x}-${focusPosition.y}`;
+    const currentTile = document.getElementById(currentId);
+
+    if (currentTile?.firstChild)
+      return currentTile.firstChild.querySelector('button').focus();
+    handleOnKeyUp({
+      code: keycode,
+      preventDefault: event.preventDefault,
+      repeat: false
+    });
+  };
+
+  useEffect(() => {
+    const isArrowKey = event => {
+      const code = event.code;
+      const right = code === 'ArrowRight';
+      const left = code === 'ArrowLeft';
+      const up = code === 'ArrowUp';
+      const down = code === 'ArrowDown';
+
+      if (!(right || left || up || down)) return false;
+      event.preventDefault();
```

Comment:
These lines of code are the same used on lines 37 - 43, i suggest to refactor the functions!

Reply:
Already refactorized.